File: December - 24/KishoreMuruganantham_first_non_repeating_char.swift
import Foundation
if let line = readLine() {
    let chars = Array(line)
    var counts = [Character: Int]()
    for c in chars {
        counts[c, default: 0] += 1
    }
    
    var found = false
    for c in chars {
        if counts[c] == 1 {
            print("The first non-repeating character is: \(c)")
            found = true
            break
        }
    }
    
    if !found {
        print("No non-repeating character found.")
    }
}

File: December - 24/KishoreMuruganantham_first_non_repeating_char.R
input <- file("stdin", "r")
s <- readLines(input, n = 1)
chars <- strsplit(s, "")[[1]]
tbl <- table(chars)

found <- FALSE
res <- ""
for (c in chars) {
  if (tbl[c] == 1) {
    found <- TRUE
    res <- c
    break
  }
}

if (found) {
  cat("The first non-repeating character is: ", res, "\n", sep = "")
} else {
  cat("No non-repeating character found.\n")
}
